test.js

```diff
@@ -394,3 +394,24 @@ test('test that missing type errors', function (t, ctx, done) {
   ws.write(data)
   ws.end()
 })
+
+test('batches according to maxBufferLength', function (t, ctx, done) {
```

Comment:
Currently we have `sourceData.length === 10`, which means we will batch `5` times with each batch having `2` items.